lib/src/async/concat.dart

```diff
@@ -40,27 +40,26 @@ class _ConcatStream<T> extends Stream<T> {
   final Iterable<Stream<T>> _streams;
 
   @override
-  StreamSubscription<T> listen(void onData(T data),
-      {Function onError, void onDone(), bool cancelOnError}) {
+  StreamSubscription<T> listen(void onData(T data)/*?*/,
+      {Function/*?*/ onError, void onDone()/*?*/, bool/*?*/ cancelOnError}) {
     cancelOnError = true == cancelOnError;
-    StreamSubscription<T> currentSubscription;
-    StreamController<T> controller;
+    StreamSubscription<T>/*?*/ currentSubscription;
+    StreamController<T>/*?*/ controller;
     final iterator = _streams.iterator;
 
-    void nextStream() {
-      bool hasNext;
+    void nextStream(StreamController<T> controller) {
+      /*late*/ bool hasNext;
```

Comment:
dunno if temp annotations support it, but this could be `late final`

Reply:
Will give it a shot in the upcoming patch that uncomments this!